apis/definition.yaml

```diff
@@ -0,0 +1,83 @@
+  name: xsecuritygroups.aws.platform.upbound.io
+spec:
+  group: aws.platform.upbound.io
+  names:
+    kind: XSecurityGroup
+    plural: xsecuritygroups
+  claimNames:
+    kind: SecurityGroup
+    plural: securitygroups
+  versions:
+    - name: v1alpha1
+      served: true
+      referenceable: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                parameters:
+                  type: object
+                  description: Parameters used to customize the cluster
+                  properties:
+                    id:
+                      type: string
+                      description: ID of this Network that other objects will use to refer to it.
+                    region:
+                      type: string
+                      description: Region is the region you'd like your resource to be created in.
+                    deletionPolicy:
+                      description: Delete the external resources when the Claim/XR is deleted. Defaults to Delete
+                      enum:
+                        - Delete
+                        - Orphan
+                      type: string
+                      default: Delete
+                    providerConfigName:
+                      description: Crossplane ProviderConfig to use for provisioning this resources
+                      type: string
+                      default: default
+                    rules:
+                      type: array
+                      items:
+                        type: object
+                        properties:
+                          type:
+                            type: string
+                            description: Type of rule, ingress (inbound) or egress (outbound)..
+                            enum:
+                              - ingress
+                              - egress
+                          fromPort:
+                            type: integer
+                          toPort:
+                            type: integer
+                          description:
+                            type: string
+                            default: Crossplane generated Security Group Rule
+                          protocol:
+                            type: string
+                          sourceSecurityGroupName:
+                            type: string
+                            description: Reference to a SecurityGroup to populate sourceSecurityGroupName
+                          cidrBlocks:
+                            type: array
+                            description: Array List of CIDR blocks (optional).
+                            items:
+                              type: string
+                          self:
+                            type: boolean
+                            description: Whether the security group itself will be added as a source to this ingress rule.
```

Comment:
aren't we missing `required` section for `rules` ? It looks like I can create broken rule with e.g. protocol with no value

Reply:
i see only an option with some CEL rules because `cidrBlocks` or `sourceSecurityGroupName` is needed or `self` without `cidrBlocks` and `sourceSecurityGroupName`

Reply:
implemented required fields and CEL validation rules 
tested with invalid manifests:

```
git:(feature/api) ✗ kubectl apply -f examples/sg-2.yaml 
The XSecurityGroup "eks-controlplane" is invalid: spec.parameters.rules[0]: Invalid value: "object": When fromPort and toPort are specified, either cidrBlocks or sourceSecurityGroupName must be set.
git:(feature/api) ✗ kubectl apply -f examples/sg-2.yaml
The XSecurityGroup "eks-controlplane" is invalid: spec.parameters.rules[2]: Invalid value: "object": When isSelf is true, neither cidrBlocks nor sourceSecurityGroupName can be specified.
```